activesupport/lib/active_support/claims.rb

```diff
@@ -0,0 +1,69 @@
+require 'active_support/time'
+
+module ActiveSupport
+  class Claims # :nodoc:
+    class InvalidClaims < StandardError; end
+    class ExpiredClaims < StandardError; end
+    
+    attr_reader :payload, :purpose, :expires_at
+    UNIVERSAL_PURPOSE = 'universal'
+
+    def initialize(payload:, **options)
+      @payload = payload
+      @purpose = self.class.pick_purpose(options)
+      @expires_at = pick_expiration(options)
+    end
+
+    class << self
+      attr_accessor :expires_in
+
+      def verify!(claims, options = {})
```

Comment:
Ah crap, actually I'm sorry. I've given you an answer instead of poking you with questions, so you could find a solution. Feel free to disregard what I've said above and come up with your own solution :smile:

My main idea behind reworking this was that the exception raising (or not) spread all over the place, and it would be tough to add extra metadata later. So finding a place to put all that junk would make the code easier to change later :smile: